packages/shared_preferences/shared_preferences_foundation/darwin/shared_preferences_foundation/Sources/shared_preferences_foundation/SharedPreferencesPlugin.swift

```diff
@@ -152,7 +152,7 @@ public class SharedPreferencesPlugin: NSObject, FlutterPlugin, UserDefaultsApi {
     var filteredPrefs: [String: Any] = [:]
     var compatiblePrefs: [String: Any] = [:]
     let allowSet = allowList.map { Set($0) }
-    if let appDomain = Bundle.main.bundleIdentifier,
+    if let appDomain = options.suiteName ?? Bundle.main.bundleIdentifier,
       let prefs = try getUserDefaults(options: options).persistentDomain(forName: appDomain)
```

Comment:
We're already getting the right domain in `getUserDefaults`; isn't the fix to call `dictionaryRepresentation` instead of `persistentDomain(...)`?

Reply:
`dictionaryRepresentation` returns keys that belong to the global domain as well.

Reply:
The docs for `persistentDomain` say:
> Calling this method is equivalent to initializing a user defaults object with [init(suiteName:)](https://developer.apple.com/documentation/foundation/userdefaults/1409957-init) passing domainName and calling the [dictionaryRepresentation()](https://developer.apple.com/documentation/foundation/userdefaults/1415919-dictionaryrepresentation) method on it.

Is that statement in the official docs incorrect?

Reply:
Yes, it's incorrect.

Reply:
https://github.com/feedback-assistant/reports/issues/165

Reply:
Please add a comment with that reference to the code so that it's clear why there is what looks like redundant code here.

Reply:
Done.